### PR TITLE
Restructure docs and host using ReadTheDocs

### PR DESCRIPTION
**Pull request by ManavalanG**
# Pull request

Please fill in the checklist below and comment as needed.

- [x] Was code modified? Briefly describe. _No_
- [x] Was documentation modified? Briefly describe. _Yes. Please see below._
- [x] Is this a bug-fix? Briefly describe. _No_
- [x] Is this a feature addition? Briefly describe. _Yes, partly. Builds doc using mkdocs and hosts using readthedocs_
- [x] Did you modify QuaC-Watch config file? If so, did you modify multiqc template
  `configs/multiqc_config_template.jinja2` and script `src/quac_watch/create_mutliqc_configs.py` as necessary? _No_
- [x] Did you perform system-level testing manually as described in master readme doc? Did it pass completely? If not why? _No_
- [x] Updated `Changelog.md` file with change logs in recommended format?

## Anything else reviewer should know?

Main goal of this PR is to restructure the docs to make it clearer to non-Cheaha users, update documentation to reflect recent CLI option changes, and host docs publicly using readthedocs. QuaC doc is now available at https://quac.readthedocs.io/, which is based on the branch in this PR.

* `Readme.md` has grown bigger and became difficult to navigate. Especially for non-Cheaha users. So documentation in readme.md is now restructured to break into multiple files and is now easier to consume.
     * Makes the doc generic to non-Cheaha users
     * Identifies parts of the docs that are specific to Cheaha or CGDS users
* Updates doc to reflect changes made in #59 
* Now hosts doc for "Sample QC review system". Its copy in MOO will be removed after this PR is complete.
* Uses [mkdocs](https://www.mkdocs.org/) to create static site for documentation.
     * Setup in this repo was used - https://github.com/ManavalanG/mana_random_notes/, which was based on https://github.com/readthedocs-examples/example-mkdocs-basic/.
* Hosts QuaC docs using [ReadTheDocs](https://readthedocs.org/) 
    * Repo is made public to make this happen.
    * Setup in this repo was used - https://github.com/ManavalanG/mana_random_notes/, which was based on https://github.com/readthedocs-examples/example-mkdocs-basic/.
* Items not directly related to docs restructuring or hosting
   * Adds github action to identify broken links in markdown files
   * Migrates Gitlab MR template to Github PR template
   * Removes html files that were part of system testing input data. They were not necessary for system testing and they heavily inflate languages-used stats in github. 
   * Adds note to readme on quac's former residence in UAB-gitlab.

### CODE REVIEW

File: docs/CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Contributing
+
+If you like to make changes to the source code, please follow these contribution guidelines.
+
+## How to contribute?
+
+<!-- markdown-link-check-disable -->
+Follow the [CGDS' best practices for code
+development](http://cgds.uab.edu/manual-of-operations/standards-definitions/#standard-practices) (see sections IDE,
```
Comment (wilkb777) [inline docs/CONTRIBUTING.md:9]: I don't think this link is accessible publicly, likely being flagged as dead consider removing.

File: docs/input_output.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+!!! tip 
+
+    Users may primarily be interested in the aggregated QC results produced by [multiqc](https://multiqc.info/),
+    both at sample-level as well as at the project-level. These multiqc reports also include summary of QuaC-Watch
+    results at the top.
+
+!!! note "CGDS users only"
+
+    QuaC's output directory structure was designed based on the output structure of the [CGDS small variant caller
+    pipeline](https://gitlab.rc.uab.edu/center-for-computational-genomics-and-data-science/sciops/pipelines/small_variant_caller_pipeline).
```
Comment (wilkb777) [inline docs/input_output.md:108]: This link isn't public, which may cause the dead link action to fail consistently. Personally I think this note can be removed completely from this section.

File: docs/system_testing.md [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+
+# activate conda env
+conda activate quac
+
+
+## use slurm or not
+# use this to submit jobs to slurm for the parent snakemake process 
+# as well as for the snakemake triggered jobs
+USE_SLURM="--snakemake_slurm --subtasks_slurm"
+# USE_SLURM=""  # uncomment and use this out if you don't want to use slurm at all
```
Comment (wilkb777) [inline docs/system_testing.md:36]:
```suggestion
# USE_SLURM=""  # uncomment this, comment out the above line, and use this if you don't want to use slurm at all
```

**Review by wilkb777 — COMMENTED** [review]
I recommend moving to adopt the contributing guidelines and code of conduct used in Rosalution for QuaC. It'll eliminate any customization or CGDS specific practices, meet community standards, and remove the linking issue I noted in the review of the current contributing docs.

File: docs/CONTRIBUTING.md (hunk 1 above)
Comment (ManavalanG) [reply docs/CONTRIBUTING.md]: Love the idea. I have now modified `contributing.md` based on rosalution's. Also added template for `issues` based on snakemake repo.

File: docs/input_output.md (hunk 2 above)
Comment (ManavalanG) [reply docs/input_output.md:108]: Love the idea. I have now modified `contributing.md` based on rosalution's. As part of this, MOO link is now removed.

Also added template for `issues` based on snakemake repo.

**Comment by ManavalanG** [comment]
@wilkb777 Ready for your review again.

Note: Github `markdown-link-check` actions still fails with this public link - https://gatk.broadinstitute.org/hc/en-us/articles/360035531972-PED-Pedigree-format. Not sure why.

**Comment by ManavalanG** [comment]
@wilkb777 Updated docs to make it clear that quac is built to use with human data.

**Review by wilkb777 — APPROVED** [review]
@ManavalanG this LGTM, ready for merge despite the link checker not handling the redirect for that one link correctly and failing the check 😂

**Pull request merged**